pkg/configuration/configuration.go

```diff
@@ -280,3 +290,74 @@ func (c *Config) KeyringPrefix() (string, error) {
 	}
 	return h, nil
 }
+
+func (c *Config) CheckForUpdate(out io.Writer, current string) error {
```

Comment:
I would suggest return Meta and write it in root, and catch write errors as done in the other cases
```suggestion
func (c *Config) CheckForUpdate(out io.Writer, current string) (Meta,error) {
```